databricks_cli/secrets/cli.py

```diff
@@ -0,0 +1,308 @@
+
+
+SCOPE_HEADER = ('Scope', 'Backend')
+SECRET_HEADER = ('Key name', 'Last updated')
+ACL_HEADER = ('Principal', 'Permission')
+VALUE_OPTIONS = ['string-value', 'bytes-value']
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help="Creates a secret scope.")
+@click.option('--scope', required=True)
+@click.option('--initial-manage-principal',
+              help='The initial principal that can manage the created secret scope.')
+@profile_option
+@eat_exceptions
+@provide_api_client
+def create_scope(api_client, scope, initial_manage_principal):
+    """
+    Creates a new secret scope with given name.
+
+    If "initial_manage_principal" is specified, the initial ACL applied to the scope is
+    applied to the supplied principal (user or group) with MANAGE permissions.
+    The only supported principal for this option is the group "users", which
+    contains all users in the workspace. If "initial_manage_principal" is not specified,
+    the initial ACL with MANAGE permission applied to the scope is assigned to the
+    API request issuer's user identity.
+    """
+    if initial_manage_principal is not None and initial_manage_principal != "users":
+        error_and_quit("The only supported principal for --initial-manage-principal is users")
+    SecretApi(api_client).create_scope(scope, initial_manage_principal)
+
+
+def _scopes_to_table(scopes_json):
+    ret = []
+    for s in scopes_json.get('scopes', []):
+        ret.append((truncate_string(s['name']), s['backend_type']))
+    return ret
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Lists all the secret scopes.')
+@click.option('--output', default=None, help=OutputClickType.help, type=OutputClickType())
+@profile_option
+@eat_exceptions
+@provide_api_client
+def list_scopes(api_client, output):
+    """
+    Lists all secret scopes available in the workspace.
+    """
+    scopes_json = SecretApi(api_client).list_scopes()
+    if OutputClickType.is_json(output):
+        click.echo(pretty_format(scopes_json))
+    else:
+        click.echo(tabulate(_scopes_to_table(scopes_json), headers=SCOPE_HEADER))
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Deletes a secret scope.')
+@click.option('--scope', required=True)
+@profile_option
+@eat_exceptions
+@provide_api_client
+def delete_scope(api_client, scope):
+    """
+    Deletes a secret scope.
+    """
+    SecretApi(api_client).delete_scope(scope)
+
+
+def _verify_and_translate_options(string_value, binary_file):
+    """
+    Translates options into actual parameters for API call.
+    Return tuple with two values representing (string_value, bytes_value).
+    """
+    if string_value and binary_file:
+        error_and_quit("At most one of {} should be provided."
+                       .format(['string-value', 'binary-file']))
+
+    if string_value is None and binary_file is None:
```

Comment:
maybe make these `if` statements `elif`'s, so we know that exactly one block gets executed.